### ISSUE TITLE
Add Experimental Data from applied-voltage to temperature time lag for continuous variable encoding model

### ISSUE THREAD

**Issue by nm19000** [schererant/uq-qnn#11]
General idea: We get discrete data from classical computations. Now we want to approximate this by continuous data and at the end of the computation on the chip discretize again.

1) We get data from Iris' measurements of applied-voltage to temperature time lag. This determines the approximation of normalized data to continuous data.

2) Write function to determine time bins for discrete classical data points from interval [a,b] with 0 <= a < b <= 1. 

3) Based on Iris measurements maximize length of bins.

4) Use uniform distribution of points in bins around classical data points from interval [a,b] with 0 <= a < b <= 1 to simulate discretely what happens on the chip during continuous computations. (Let number of uniformly sampled points in bins around classical data points get very large to approximate continuity)

Do this here in the simulation https://github.com/schererant/uq-qnn/blob/47fc6f66ebb3cb7d6836cd00cc355534390bc7b2/src/fastencodingmodel.py#L234

5) obtain chip parameters for x^4 and test with new chip parameters in experiment.

**Comment by schererant** [comment]
Taking the measurements and creating time bins, following [https://github.com/schererant/uq-qnn/blob/continuousencoding/notebooks/07_bins.ipynb](url)


![Image](https://github.com/user-attachments/assets/acf60bfb-cfad-40cb-88d5-7cf72bfee3c3)

Total number of bins: 20
Total number of measurements: 674
Measurement mean value: 1.6155
Measurement standard deviation: 0.6612
Applied uncertainty (±15.0%): ±0.0992
Sampling time resolution: 0.001s

**Comment by nm19000** [comment]
Somehow the notebook is not accessible here [https://github.com/schererant/uq-qnn/blob/continuousencoding/notebooks/07_bins.ipynb](https://github.com/schererant/uq-qnn/issues/url)

**Comment by nm19000** [comment]
For testing the optimization of circuit parameters of the continuous encoding (which is only possible on the actual hardware), we need to edit the training procedure. One approach could be:

1)  In https://github.com/schererant/uq-qnn/blob/04fa3e4a00ccbd0c41ca17e6def256a491b97d83/src/fastencodingmodel.py#L110 we need to set the encoded phases to the mean values of the phases of each time bin  (or mean of phase bins).
2) In https://github.com/schererant/uq-qnn/blob/04fa3e4a00ccbd0c41ca17e6def256a491b97d83/src/fastencodingmodel.py#L128 (maybe?) make a nested loop where each index i corresponds to a bin.
3) Then, in a nested loop (or parallel, but this causes problems with the memristor) for each bin, based on the (time/phase) bin boundary values of phases sample $s \in \mathbb{N}$ points and allocate circuit predictions. For each bin compute the mean of the allocated s circuit predictions.
4) Adapt the loss in https://github.com/schererant/uq-qnn/blob/04fa3e4a00ccbd0c41ca17e6def256a491b97d83/src/fastencodingmodel.py#L162 from 

$$ \frac{1}{ms}\sum_{i=1}^{ms} \abs{y_train[i] - probstate_001[i]}^2,$$

to 

$$ \frac{1}{m}\sum_{i=1}^{m} \frac{1}{s} \sum_{k=1}^s \abs{y_train[i] - probstate_001[i,k]}^2,$$

where $m \in \mathbb{N}$ is the number of bins (so also the number of data points) and $s \in \mathbb{N}$ is the number of samples of phases per bin. Moreover for each bin $I \in \{1,...,m\}$  probstate_001[i,k] is the quantity computed in part 3).

**Comment by nm19000** [comment]
Comment: for initial experimental tests we need to input the above mentioned bin mean values (with the inverse of this function https://github.com/schererant/uq-qnn/blob/04fa3e4a00ccbd0c41ca17e6def256a491b97d83/src/model.py#L108), instead of the input data in https://github.com/schererant/uq-qnn/blob/04fa3e4a00ccbd0c41ca17e6def256a491b97d83/src/dataloader.py#L52 to the function we want to fit (e.g. $f(x) = x^4$ in https://github.com/schererant/uq-qnn/blob/04fa3e4a00ccbd0c41ca17e6def256a491b97d83/src/dataloader.py#L9) .
